libs/ledger-live-common/src/hw/connectApp.ts

```diff
@@ -496,11 +497,16 @@ const isDmkTransport = (
 };
 
 const appNameToDependency = (appName: string): ApplicationDependency => {
-  let constraints: ApplicationConstraint[] | undefined = undefined;
-  const minVersion = getMinVersion(appName);
-  if (minVersion !== undefined) {
-    constraints = [{ minVersion: minVersion as ApplicationVersionConstraint }];
-  }
+  const constraints = Object.values(DeviceModelId).reduce((result, model) => {
+    const minVersion = getMinVersion(appName, model);
+    if (minVersion) {
+      result.push({
+        minVersion: minVersion as ApplicationVersionConstraint,
+        applicableModels: [model],
+      });
+    }
+    return result;
+  }, [] as ApplicationConstraint[]);
```

Comment:
Maybe use `reduce<ApplicationConstraint[]>(...)`, slightly cleaner than using this `[] as ApplicationConstraint[]`